clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1045,10 +1046,14 @@ public void onComplete(Map<TopicPartition, OffsetAndMetadata> offsets, Exception
         if (subscriptions.partitionsAutoAssigned()) {
             generation = generation();
             // if the generation is null, we are not part of an active group (and we expect to be).
-            // the only thing we can do is fail the commit and let the user rejoin the group in poll()
+            // the only thing we can do is fail the commit and let the user rejoin the group in poll();
+            // we use RebalanceInProgressException to indicate this may not be a fatal error, and leave
+            // it to users whether they want to handle differently than CommitFailed
             if (generation == null) {
                 log.info("Failing OffsetCommit request since the consumer is not part of an active group");
-                return RequestFuture.failure(new CommitFailedException());
+                return RequestFuture.failure(new RebalanceInProgressException("Offset commit cannot be completed since the " +
```

Comment:
This is the main proposal for returning a different exception.

Reply:
Is it always the case that if `generation` is `null` that it's due to a rebalance in progress? Or do we want to keep the `ComitFailedException` and add to the exception message that the exception could be due to a rebalance?

Reply:
This is a good question, I would replace this case with the RetriableCommitFailed exception (see my other comment below).